src/components/bottomSheet/bottom-sheet.js

```diff
@@ -189,16 +197,15 @@ function MdBottomSheetProvider($$interimElementProvider) {
         $animate.enter(backdrop, options.parent, null);
       }
 
-      var bottomSheet = new BottomSheet(element, options.parent);
-      options.bottomSheet = bottomSheet;
+      options.cleanupGestures = options.isLockedOpen ? angular.noop : new BottomSheetGestures(element, options.parent);
```

Comment:
It doesn't make any sense to have a noop function here. Check for example the dialog interim element / component.

Regarding API, the options are not any API for the developers. At time of `show` those are just internal and used to store things between the different states.